compiler/src/parser.rs

```diff
@@ -453,4 +463,16 @@ mod tests {
             r#"[Class(ClassSource { qualifiers: Qualifiers([]), name: "Test", base: None, members: [Field(FieldSource { declaration: Declaration { annotations: [], qualifiers: Qualifiers([Private]), name: "m_field", pos: Pos(114) }, type_: TypeName { name: Owned("String"), arguments: [] } })], pos: Pos(13) })]"#
         );
     }
+
+    #[test]
+    fn parse_escaped_string() {
+        let escaped = lang::escaped_string(
+            r#""This is a backslash \'\\\' \"escaped\" string \u{03BB}""#
+        ).unwrap();
+
+        assert_eq!(
+            escaped,
+            r#"This is a backslash '\' "escaped" string λ"#
```

Comment:
I'm much happier with this test, it targets just my specific addition and doesn't rely on magic debug formatting